pyiceberg/table/__init__.py

```diff
@@ -801,7 +801,7 @@ def name(self) -> Identifier:
         Returns:
             An Identifier tuple of the table name
         """
-        return self.identifier
+        return self._identifier
```

Comment:
I think this change is problematic since we change the contract here. Everyone that's using 8.0.0 expects the catalog to be part of the name, with his change we remove that one without letting the user know.

Reply:
Yes, that would be the correct way. I don't see any other option to just accept this. Since the `.name()` is just new, I think it is acceptable, but we probably should call it out at the release.

Reply:
Deprecating it seems like a lot of work, and will also be confusing to the end-user. So I don't think we have many options, and I think what you're suggesting in this PR is the best path forward. Getting this out ASAP in 0.8.1 would be great 👍

Reply:
sounds good, make sense to me!